diff --git a/src/compositor/compositor.cpp b/src/compositor/compositor.cpp
index dea25ca..80379ba 100644
--- a/src/compositor/compositor.cpp
+++ b/src/compositor/compositor.cpp
@@ -157,6 +157,8 @@ Compositor::Compositor(const QString &socket)
     : QWaylandQuickCompositor(socket.isEmpty() ? 0 : qPrintable(socket), WindowManagerExtension | OutputExtension | QtKeyExtension | TouchExtension | HardwareIntegrationExtension | SubSurfaceExtension)
     , d_ptr(new CompositorPrivate(this))
 {
+    //TODO: setting actually the correct orientation
+    QWaylandQuickCompositor::setScreenOrientation(Qt::PortraitOrientation);
     qRegisterMetaType<QuickSurface *>("QuickSurface*");
     qRegisterMetaType<Output *>("Output*");
     qRegisterMetaType<WindowView *>("WindowView*");
